### ISSUE TITLE
dbt cli stopped working after upgrading to 1.8.0

### ISSUE THREAD

**Issue by thijs-nijhuis-shell** [databricks/dbt-databricks#681]
### Describe the bug
Once I upgrade to dbt-databricks 1.8.0 none of the dbt cli command work anymore and all fail with the same error message. For instance, if I run `dbt -h` or `dbt deps` I get this error:
```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\Thijs.Nijhuis\OneDrive - Shell\Documents\git\Template-EDP-DBT-Reference-Pipeline\.venv\Scripts\dbt.exe\__main__.py", line 4, in <module>
  File "C:\Users\Thijs.Nijhuis\OneDrive - Shell\Documents\git\Template-EDP-DBT-Reference-Pipeline\.venv\Lib\site-packages\dbt\cli\__init__.py", line 1, in <module>
    from .main import cli as dbt_cli  # noqa
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Thijs.Nijhuis\OneDrive - Shell\Documents\git\Template-EDP-DBT-Reference-Pipeline\.venv\Lib\site-packages\dbt\cli\main.py", line 14, in <module>
    from dbt.cli import requires, params as p
  File "C:\Users\Thijs.Nijhuis\OneDrive - Shell\Documents\git\Template-EDP-DBT-Reference-Pipeline\.venv\Lib\site-packages\dbt\cli\requires.py", line 10, in <module>
    from dbt.adapters.factory import adapter_management, register_adapter, get_adapter
```

Once I install dbt-databricks 1.7.16, the problem is solved.

### Steps To Reproduce
Simply pip install 1.8.0 and run any cli command.

### Expected behavior
The dbt cli to still work after install dbt-databricks

### Screenshots and log output
See bug description

### System information
**The output of `dbt --version`:**
I cannot run this command as that gives me the same error. When I run pip freeze I see theis (among other packages)
```
databricks-sdk==0.17.0
databricks-sql-connector==3.1.2
dbt-adapters==1.1.1
dbt-common==1.1.0
dbt-core==1.8.1
dbt-databricks==1.8.0
dbt-extractor==0.5.1
dbt-semantic-interfaces==0.5.1
dbt-spark==1.8.0
```

**The operating system you're using:**
'Windows 11 enterprise' but I get the exact same error when running from a GitHub action on ubuntu.

**The output of `python --version`:**
Python 3.11.4

### Additional context
n.a.

**Comment by thijs-nijhuis-shell** [comment]
In addition, I also pip installed the 'prep_for_1.8.1b1' branch locally but that gave me the same error.

**Comment by benc-db** [comment]
What's the actual error?  You start at the stack trace, but what is the exception?  Also, can you share full pip freeze?

**Comment by thijs-nijhuis-shell** [comment]
Thanks for pointing me tot that thread @benc-db . Think that is indeed the exact same problem.

So what I understand from it, is that this only happens when upgrading between a dbt-core pre 1.8.0 version to dbt-core 1.8.0 or later. The reason seems to be that the dbt-adapters package is now a standalone package but still resided in the dbt-core folder will which pip can't (always) handle.

For me, running this command, which I got from that thread, solved the issue (although it does throw an error):
```shell
pip install --force-reinstall dbt-adapters
```

Another way, would be to complete uninstall dbt-databricks and dbt-core and than clean install 1.8.0 or higher.

Closing this thread as this has nothing to with dbt-databricks but is a dbt-core issue.
